### DISCUSSION TITLE
Is there any ways to use custom file as entrypoint?

### DISCUSSION THREAD

**Discussion by aciddust** [deta/cloud-docs#520] [Help] [answered]
![image](https://user-images.githubusercontent.com/18468557/194794896-25a710cb-6aab-479e-901a-e2ea2ca0802a.png)
> I wanna use app.main as entrypoint 

It seems that Deta's python service is currently using `main.py` in the root location as an entry point.

Is there a way to use another file as an entry point?

**Comment by AyushSehrawat** [comment, marked as answer]
Currently there's no way for micros having custom entrypoint, but they will be there in Deta space if not yet implemented

**Reply by AyushSehrawat** [reply]
https://alpha.deta.space

**Reply by aciddust** [reply]
Thanks again mate

**Reply by AyushSehrawat** [reply]
But you can try playing with main.py so that it runs your main app file...Personally never tried

**Reply by aciddust** [reply]
Works great when deploying main.py in root path.

- [image steganography app](https://github.com/dev4hobby/image-steganography)
  - [API w/ Deta] (https://binjector.deta.dev/)

I moved my app from `Heroku` to `Deta` last night
it works nicely

**Reply by aciddust** [reply]
![image](https://user-images.githubusercontent.com/18468557/194795959-a1c270c5-3273-47fc-9c17-678f5c11ca59.png)
![image](https://user-images.githubusercontent.com/18468557/194796034-60a7daae-f310-4b89-b64c-0a665b348a45.png)

state file has been updated
and deployd to [https://uajlrj.deta.dev/](https://uajlrj.deta.dev/)

but default app is running 

maybe I think they are generate default `main.py` if not exists

**Reply by AyushSehrawat** [reply]
```
.deta ( folder) 
main.py ( import and runs from the main file in app
app ( folder) 
==== your app folder content
```

**Reply by AyushSehrawat** [reply]
Did you implement something like this?

**Reply by AyushSehrawat** [reply]
> ![image](https://user-images.githubusercontent.com/18468557/194795959-a1c270c5-3273-47fc-9c17-678f5c11ca59.png) ![image](https://user-images.githubusercontent.com/18468557/194796034-60a7daae-f310-4b89-b64c-0a665b348a45.png)
> 
> state file has been updated and deployd to https://uajlrj.deta.dev/
> 
> but default app is running
> 
> maybe I think they are generate default `main.py` if not exists

If i remember correctly if no main.py is found it gets generated with Hello, World... Not 100% sure

**Reply by aciddust** [reply]
![image](https://user-images.githubusercontent.com/18468557/194796546-364e527a-2ac6-42f0-aa1d-92888ee0ec39.png)

no `main.py` here

> If i remember correctly if no main.py is found it gets generated with Hello, World... Not 100% sure

yes that's what I'm having now

**Reply by aciddust** [reply]
deployed to [https://uajlrj.deta.dev/](https://uajlrj.deta.dev/)

this site shows `Hello, World!`

**Reply by AyushSehrawat** [reply]
Hey to see what's happing try,

```
deta clone --name micro_name --project project_name /tmp
```

Remove `--project project_name` if using default project

**Reply by AyushSehrawat** [reply]
> deployed to https://uajlrj.deta.dev/
> 
> this site shows `Hello, World!`

Try setting up a main.py file in root directory which imports the app from the main file in app folder...

Then do some stuff and run it using `uvicorn main:app`, if it works like that then deploy to deta

Note : uvicorn main:app is not the command deta uses but it's something similar to it

**Reply by aciddust** [reply]
> Hey to see what's happing try,
> ```bash
> deta clone --name micro_name --project project_name /tmp
> ```
> Remove `--project project_name` if using default project

![image](https://user-images.githubusercontent.com/18468557/194796970-f1839308-e8ec-40f4-b7ed-b106127d5003.png)

`main.py` generated (including 'Hello, world!')

**Reply by aciddust** [reply]
> > deployed to https://uajlrj.deta.dev/
> > this site shows `Hello, World!`
> 
> Try setting up a main.py file in root directory which imports the app from the main file in app folder...
> 
> Then do some stuff and run it using `uvicorn main:app`, if it works like that then deploy to deta
> 
> Note : uvicorn main:app is not the command deta uses but it's something similar to it

A'ight
I'll try with main.py located in root

Many thanks again.

**Reply by lemonyte** [reply]
You can always create a `main.py` file in the root directory and import your ASGI/WSGI app instance from wherever your actual code is located.

```python
# main.py
from subdir.actualfile import app
```
